plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryClientFactory.java

```diff
@@ -37,6 +37,7 @@
     private final IdentityCacheMapping identityCacheMapping;
     private final BigQueryCredentialsSupplier credentialsSupplier;
     private final Optional<String> parentProjectId;
+    private final Optional<String> projectId;
```

Comment:
Currently I was unable to figure out difference. Seems the "parent" is used for billing purposes but I haven't been able to verify since the JavaDoc for the client doesn't explain it and it's kind of hard to observe it on BigQuery dashboard.

Reply:
Hello! Right, `parentProjectId` if set will be used as the "billing" project, the one incurring the cost. What that means in practice is that `parentProjectId` or Service Account projectId should be used to create the underlying read session. Creating this session requires special perm `BigQuery Read Session User` and normally should be granted on the request project which will be the one owning the session. I hope this clarifies things a bit more.